common/common-rest/src/main/java/io/servicecomb/common/rest/RestProducerInvocation.java

```diff
@@ -65,115 +52,26 @@ public void invoke(Transport transport, HttpServletRequestEx requestEx, HttpServ
-        restOperationMeta.getOperationMeta(),
-        args);
-    this.invocation.setMetricsData(metricsData);
-    updateMetrics();
-    invoke();
-  }
-
-  protected RestOperationMeta findRestOperation() {
+  protected void findRestOperation() {
     String targetMicroserviceName = requestEx.getHeader(Const.TARGET_MICROSERVICE);
     if (targetMicroserviceName == null) {
       // for compatible
       targetMicroserviceName = RegistryUtils.getMicroservice().getServiceName();
     }
     MicroserviceMeta selfMicroserviceMeta =
         CseContext.getInstance().getMicroserviceMetaManager().ensureFindValue(targetMicroserviceName);
-    ServicePathManager servicePathManager = ServicePathManager.getServicePathManager(selfMicroserviceMeta);
-    if (servicePathManager == null) {
-      LOGGER.error("No schema in microservice");
-      throw new InvocationException(Status.NOT_FOUND, Status.NOT_FOUND.getReasonPhrase());
-    }
-
-    OperationLocator locator =
-        servicePathManager.producerLocateOperation(requestEx.getRequestURI(), requestEx.getMethod());
-    requestEx.setAttribute(RestConst.PATH_PARAMETERS, locator.getPathVarMap());
-
-    return locator.getOperation();
+    findRestOperation(selfMicroserviceMeta);
   }
 
   @Override
-  protected void doInvoke() throws Throwable {
-    invocation.next(resp -> {
-      sendResponseQuietly(resp);
-      endMetrics();
-
-    });
-  }
-
-  /**
-   * Init the metrics. Note down the queue count and start time.
-   * @param operationMeta Operation data
-   * @return QueueMetrics
-   */
-  private QueueMetrics initMetrics(OperationMeta operationMeta) {
-    QueueMetrics metricsData = new QueueMetrics();
-    metricsData.setQueueStartTime(System.currentTimeMillis());
-    metricsData.setOperQualifiedName(operationMeta.getMicroserviceQualifiedName());
-    QueueMetricsData reqQueue = MetricsServoRegistry.getOrCreateLocalMetrics()
-        .getOrCreateQueueMetrics(operationMeta.getMicroserviceQualifiedName());
-    reqQueue.incrementCountInQueue();
-    return metricsData;
-  }
-
-  /**
-   * Update the queue metrics.
-   */
-  private void updateMetrics() {
-    QueueMetrics metricsData = (QueueMetrics) this.invocation.getMetricsData();
-    if (null != metricsData) {
-      metricsData.setQueueEndTime(System.currentTimeMillis());
-      QueueMetricsData reqQueue = MetricsServoRegistry.getOrCreateLocalMetrics()
-          .getOrCreateQueueMetrics(restOperationMeta.getOperationMeta().getMicroserviceQualifiedName());
-      reqQueue.incrementTotalCount();
-      Long timeInQueue = metricsData.getQueueEndTime() - metricsData.getQueueStartTime();
-      reqQueue.setTotalTime(reqQueue.getTotalTime() + timeInQueue);
-      reqQueue.setMinLifeTimeInQueue(timeInQueue);
-      reqQueue.setMaxLifeTimeInQueue(timeInQueue);
-      reqQueue.decrementCountInQueue();
-    }
+  protected OperationLocator locateOperation(ServicePathManager servicePathManager) {
```

Comment:
What will happen if we cannot find the OperationLocator?

Reply:
if locate failed, throw 404 or 405 exception.